src/main/java/ru/tyshchenko/infosearch/utils/ParserUtils.java

```diff
@@ -14,4 +20,17 @@ public static Document getDocumentPage(String url) {
                         "(KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36")
                 .get();
     }
+    @SneakyThrows
+    public static String getSourceText(Path filePath) {
+        return Jsoup.parse(Files.readString(filePath)).text();
+    }
+
+    @SneakyThrows
+    public static String getSourceTextFromFiles(Path dirPath) {
+        try(Stream<Path> walk = Files.walk(dirPath)) {
+            return walk.filter(Files::isRegularFile)
+                    .map(ParserUtils::getSourceText)
+                    .reduce("", (a, b) -> a + b);
```

Comment:
Hello Ural, let's try the next code:
var result = new StringBuilder();
//instead of reduce lets try forEach
.forEach(result::append);
you can try it make parallel, then please use StringBuffer instead of StringBuilder, cause it is thread-safe